### PR TITLE
Add Nix files and commit `Cargo.lock`

### PR DESCRIPTION
**Pull request by pnmadelaine**
Add a Nix expression to build `ml-kem`

### CODE REVIEW

**Comment by pnmadelaine** [comment]
I disabled LTO in the Nix build as it causes the build to fail. Am I missing something, is there a specific toolchain to use? I am using `gcc` and `ld`.

**Comment by pnmadelaine** [comment]
using mold instead of ld seems to fix the issue!

File: libcrux-ml-kem/c/CMakeLists.txt [hunk 1]
```diff
@@ -108,13 +108,7 @@ target_link_libraries(sha3_test PRIVATE
 
 # --- Benchmarks
 
-FetchContent_Populate(benchmark
```
Comment (franziskuskiefer) [inline libcrux-ml-kem/c/CMakeLists.txt:111]: Where does this change come from? It doesn't look right.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks, generally looks good except for that one change.

**Comment by pnmadelaine** [comment]
@franziskuskiefer I'll just continue the conversation here as github seems to double all my comments on the code (?)

I think you meant using `fetch_package` was not right because it required installing `benchmark` manually? I used `FetchContent_Declare` instead, is that okay with you?

**Comment by W95Psp** [comment]
(quoting myself from https://github.com/cryspen/bertie/pull/118, since it's the same concern)
> I see no point in not committing flake.lock, of we don't commit flake.lock, we should drop the nix expressions altogether. The nix job you added will check the nix expressions are up to date as code changes, that's all we need imo. Locked flakes are especially useful to make sure we'll be able to build the project in the future.

**Comment by pnmadelaine** [comment]
converting to draft as this is currently broken after updating

**Review by protz — APPROVED** [review]

**Review by franziskuskiefer — APPROVED** [review]

**Pull request merged**